relay-server/src/endpoints/playstation.rs

```diff
@@ -0,0 +1,153 @@
+use axum::extract::{DefaultBodyLimit, Request};
+use axum::response::IntoResponse;
+use axum::routing::{post, MethodRouter};
+use axum::RequestExt;
+use bytes::Bytes;
+use lz4_flex::frame::FrameDecoder as lz4Decoder;
+use multer::Multipart;
+use relay_config::Config;
+use relay_dynamic_config::Feature;
+use relay_event_schema::protocol::EventId;
+use std::io::Cursor;
+use std::io::Read;
+
+use crate::endpoints::common::{self, BadStoreRequest, TextResponse};
+use crate::envelope::ContentType::OctetStream;
+use crate::envelope::{AttachmentType, Envelope};
+use crate::extractors::{RawContentType, Remote, RequestMeta};
+use crate::service::ServiceState;
+use crate::utils;
+
+/// The extension of a prosperodump in the multipart form-data upload.
+const PROSPERODUMP_EXTENSION: &str = "prosperodmp";
+
+/// The extension of a screenshot in the multipart form-data upload.
+const SCREENSHOT_EXTENSION: &str = "jpg";
+
+/// The extension of a video in the multipart form-data upload.
+const VIDEO_EXTENSION: &str = "webm";
+
+/// The extension of a memorydump in the multipart form-data upload.
+const MEMORYDUMP_EXTENSION: &str = "prosperomemdmp";
+
+/// Prosperodump attachments should have these magic bytes
+const PROSPERODUMP_MAGIC_HEADER: &[u8] = b"\x7FELF";
+
+/// Magic bytes for lz4 compressed prosperodump containers.
+const LZ4_MAGIC_HEADER: &[u8] = b"\x04\x22\x4d\x18";
+
+fn validate_prosperodump(data: &[u8]) -> Result<(), BadStoreRequest> {
+    if !data.starts_with(PROSPERODUMP_MAGIC_HEADER) {
+        relay_log::trace!("invalid prosperodump file");
+        return Err(BadStoreRequest::InvalidProsperodump);
+    }
+
+    Ok(())
+}
+
+// TODO: Decide if we want to move this into a utils since it is duplicate.
+/// Convenience wrapper to let a decoder decode its full input into a buffer
+fn run_decoder(decoder: &mut Box<dyn Read>) -> std::io::Result<Vec<u8>> {
+    let mut buffer = Vec::new();
+    decoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
```

Comment:
We **must** limit the amount of bytes we decompress (read zip bomb).

Check how other places in relay determine the limit.